pkg/clusteragent/autoscaling/workload/controller.go

```diff
@@ -272,8 +280,20 @@ func (c *Controller) syncPodAutoscaler(ctx context.Context, key, ns, name string
 	// Reaching this point, we had an error in processing, clearing up global error
 	podAutoscalerInternal.SetError(nil)
 
-	// Now that everything is synced, we can perform the actual processing
-	result, err := c.handleScaling(ctx, podAutoscaler, &podAutoscalerInternal)
+	isAdded := c.AutoscalerHeap.InsertIntoHeap(autoscaling.TimestampKey{
+		Timestamp: podAutoscalerInternal.CreationTimestamp(),
+		Key:       key,
+	})
+
+	result := autoscaling.NoRequeue
+	var err error
+	if !isAdded {
+		// Number of DatadogPodAutoscaler objects exceeds the limit
+		podAutoscalerInternal.SetError(fmt.Errorf("Too many DatadogPodAutoscaler objects created, ignoring this one %s", key))
+	} else {
+		// Now that everything is synced, we can perform the actual processing
+		result, err = c.handleScaling(ctx, podAutoscaler, &podAutoscalerInternal)
```

Comment:
It'd be nice to not have the main logic in an `else` though it would require a to change a bit the code structure.

Reply:
Makes sense, I can wait for the changes from https://github.com/DataDog/datadog-agent/pull/28723 https://github.com/DataDog/datadog-agent/blob/f35a065b1e4c9f735ea00edceba0718f249bbe5a/pkg/clusteragent/autoscaling/workload/controller.go#L413-L427
to be merged so I can make a status update and return early here